pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdSinks.java

```diff
@@ -368,16 +419,23 @@ protected FunctionDetails createSinkConfig(SinkConfig sinkConfig) throws IOExcep
             // check if configs are valid
             validateSinkConfigs(sinkConfig);
 
-            String sinkClassName = ConnectorUtils.getIOSinkClass(sinkConfig.getArchive());
+            String sinkClassName = null;
+            String typeArg = null;
+
+            FunctionDetails.Builder functionDetailsBuilder = FunctionDetails.newBuilder();
+
+            boolean isBuiltin = sinkConfig.getArchive().startsWith(Utils.BUILTIN);
 
-            String typeArg;
-            try (NarClassLoader ncl = NarClassLoader.getFromArchive(new File(sinkConfig.getArchive()),
-                    Collections.emptySet())) {
-                typeArg = sinkConfig.getArchive().startsWith(Utils.FILE) ? null
-                        : Utils.getSinkType(sinkClassName, ncl).getName();
+            if (!isBuiltin) {
+                sinkClassName = ConnectorUtils.getIOSinkClass(sinkConfig.getArchive());
+
+                try (NarClassLoader ncl = NarClassLoader.getFromArchive(new File(sinkConfig.getArchive()),
```

Comment:
at line #434, if archive is "file" url then we don't need to access nar at line#432. so, should this line come after line#434?

also here, we are assuming that archive packaging will be always `file://` or nar for connector. so, should we document it at `--archive option`

Reply:
Ideally the preferred form here would be to submit the "builtin" connectors -- Archive will be left to submit custom built connectors. We'll update the docs overall to reflect that before the release.